src/main/java/prefs/ConfigFile.java

```diff
@@ -0,0 +1,136 @@
+        this.configDirectory = configDirectory;
+
+        setupGson();
+        createDirectoryIfNotExist();
+    }
+
+    /**
+     * configure the gson object that will serialise and deserialise json data
+     */
+    private void setupGson() {
+        gson = new GsonBuilder()
+                .setPrettyPrinting()
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonSerializer<LocalDateTime>) (src, typeOfSrc, context) -> {
+                            Instant instant = src.atZone(ZoneId.systemDefault()).toInstant();
+                            long epochMilli = instant.toEpochMilli();
+                            return new JsonPrimitive(epochMilli);
+                        })
+                .registerTypeAdapter(LocalDateTime.class,
+                        (JsonDeserializer<LocalDateTime>) (json, typeOfT, context) -> {
+                            Instant instant = Instant.ofEpochMilli(json.getAsJsonPrimitive().getAsLong());
+                            return LocalDateTime.ofInstant(instant, ZoneId.systemDefault());
+                        }
+                ).create();
+    }
+
+    /**
+     * Attempts to create the config directory if it does not exist.
+     */
+    private void createDirectoryIfNotExist() {
+        File directory = new File(configDirectory);
+        boolean directoryExists = directory.exists() && directory.isDirectory();
+        if (!directoryExists) {
+            if (directory.mkdirs()) {
+                logger.info("Config directory created: " + directory.toString());
+            } else {
+                logger.warn("Could not create config file directory");
+            }
+        }
+    }
+
+    /**
+     * Returns the File representation at the config file path
+     */
+    private File getConfigFile() {
+        return new File(configDirectory, configFileName).getAbsoluteFile();
+    }
+
+    //--- Public methods
+
+    /**
+     * Saves the config object to a file.
+     * @param config The config object to be saved.
+     */
+    public void saveConfig(Config config) {
+        File configFile = getConfigFile();
+        try (Writer writer = new OutputStreamWriter(new FileOutputStream(configFile), CHARSET)) {
+            gson.toJson(config, config.getClass(), writer);
+            writer.close();
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+        }
+    }
+
+    /**
+     * Loads a config object from a config file.
+     * @param config The config object of the type to load
+     * @return A new config object that contains the data in the config file.
+     */
+    public Config loadConfig(Config config) {
+        File configFile = getConfigFile();
+        try {
+            if (!configFile.exists()) {
+                boolean fileCreated = configFile.createNewFile();
+                if (!fileCreated) {
+                    logger.warn("Failed to create file!" + configFile.toString());
+                }
```

Comment:
Should we `return` or throw exception here? Since if the file cannot be created, we shouldn't attempt to read the file.